e2e/testdata/otel-collector/otel-cm-full-gcm.yaml

```diff
@@ -0,0 +1,326 @@
+# Copyright 2022 Google LLC
```

Comment:
Do we need this as a test artifact? Can it be generated from code so we make sure it's the same as what a user gets? Or is there something different about it for testing?

Reply:
This ConfigMap is a bit different where it takes off the metric filtering and exports all applicable metric to GCM to test the overall behavior, it's definitely different from what the user would get. The user faced behavior is tested in `TestOtelCollectorDeployment`.

Reply:
> it takes off the metric filtering and exports all applicable metric to GCM

Do we need to test the metrics that are normally filtered?

If not, we can just use the same ConfigMap the customer gets, right?

Reply:
In theory we supports viewing all the metrics we listed in the public doc in GCM if user is experienced enough to adjust the filter configuration, so it's better to make sure they all work in this test.

Reply:
Is there any way we can make this into a feature flag that we can enable dynamically, instead of hard-coding another permutation here?

I'm trying to avoid needing to manually keep them in sync.

Reply:
It's possible to generate test ConfigMap and patch the data field as needed

Reply:
Will clean up in another PR.